lib/cisco_node_utils/ntp_server.rb

```diff
@@ -66,20 +64,41 @@ def ==(other)
 
     def create
       config_set('ntp_server', 'server', state: '', ip: @ntpserver_id,
-                  prefer: @ntpserver_prefer ? 'prefer' : '')
+                  prefer: @prefer == 'true' ? 'prefer' : '',
+                  vrf: @vrf ? "use-vrf #{@vrf}" : '',
+                  key: @key ? "key #{@key}" : '',
+                  minpoll: @minpoll ? "minpoll #{@minpoll}" : '',
+                  maxpoll: @maxpoll ? "maxpoll #{@maxpoll}" : '')
     end
 
     def destroy
```

Comment:
I believe you may be able to simplify the `destroy` method and only specify the `@ntpserver_id`. I did some quick testing and I was able to remove the config without specifying any of the other parameters.

`config_set('ntp_server', state: 'no', ip: @ntpserver_id)`